lsp-lua.el

```diff
@@ -0,0 +1,156 @@
+;;; lsp-lua.el --- description -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020 E. Alexander Barbosa
+
+;; Author: E. Alexander Barbosa <elxbarbosa@outlook.com>
+;; Keywords:
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; LSP Lua clients
+
+;;; Code:
+
+(require 'lsp-mode)
+(require 'f)
+
+(defgroup lsp-emmy-lua nil
+  "LSP support for EmmyLua Language Server."
+  :group 'lsp-mode
+  :version "7.1"
+  :link '(url-link "https://github.com/EmmyLua/EmmyLua-LanguageServer"))
+
+(defcustom lsp-emmy-lua-java-path (executable-find "java")
+  "Path to Java Runtime."
+  :group 'lsp-emmy-lua
+  :version "7.1"
+  :risky t
+  :type 'file)
+
+(defcustom lsp-emmy-lua-jar-path (f-join lsp-server-install-dir "EmmyLua-LS-all.jar")
+  "Path EmmyLua language server jar file."
+  :group 'lsp-emmy-lua
+  :version "7.1"
+  :risky t
+  :type 'file)
+
+(defcustom lsp-emmy-lua-args '("-jar")
+  "List of arguments given to Lua Language server."
+  :group 'lsp-emmy-lua
+  :version "7.1"
+  :type  '(repeat string))
+
+(defcustom lsp-emmy-lua-command `(,lsp-emmy-lua-java-path
+                                  ,@lsp-emmy-lua-args
+                                  ,lsp-emmy-lua-jar-path)
+  "List of arguments given to Lua Language server."
+  :group 'lsp-emmy-lua
+  :version "7.1"
+  :type '(repeat string))
+
+(defun lsp-emmy-lua-test ()
+  "Test Emmy Lua."
+  (not (and
+        (f-exists? lsp-emmy-lua-java-path)
+        (f-exists? lsp-emmy-lua-jar-path))))
+
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection lsp-emmy-lua-command (lsp-emmy-lua-test))
```

Comment:
the second argument of lsp-emmu-lua-test is function, so this should be 'lsp-emmy-lua-test instead of (lsp-emmy-lua-test). Fix it here and on a few more places in the review.

Reply:
done in all clients